src/main/java/domain/Payment.java

```diff
@@ -0,0 +1,31 @@
+package domain;
+
+import java.util.Arrays;
+
+public enum Payment {
+	CARD(1, price -> price),
+	CASH(2, price -> (int)(0.95 * price));
+
+	private final int number;
+	private final PricingDiscountStrategy pricingDiscountStrategy;
+
+	Payment(int number, PricingDiscountStrategy pricingDiscountStrategy) {
+		this.number = number;
+		this.pricingDiscountStrategy = pricingDiscountStrategy;
+	}
+
+	public static Payment findByNumber(int number) {
+		return Arrays.stream(values())
+			.filter(payment -> payment.number == number)
+			.findAny()
+			.orElseThrow(() -> new IllegalArgumentException("해당하는 결제 수단이 존재하지 않습니다."));
```

Comment:
```suggestion
		return Arrays.stream(values())
			.filter(payment -> payment.number == number)
			.findAny()
			.orElseThrow(() -> new IllegalArgumentException("해당하는 결제 수단이 존재하지 않습니다. number=" + number));
```
상세한 exception을 남겨볼까요?

Reply:
또 이런 실수가... 수정했습니다.